plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java

```diff
@@ -8,55 +8,42 @@
 import net.mcjukebox.plugin.bukkit.managers.RegionManager;
 import net.mcjukebox.plugin.bukkit.managers.shows.Show;
 import net.mcjukebox.plugin.bukkit.managers.shows.ShowManager;
-import net.mcjukebox.shared.utils.RegionUtils;
-import net.mcjukebox.shared.api.Region;
+import org.bukkit.Location;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.PlayerJoinEvent;
 import org.bukkit.event.player.PlayerMoveEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.event.player.PlayerTeleportEvent;
 import org.bukkit.event.vehicle.VehicleMoveEvent;
+import org.codemc.worldguardwrapper.WorldGuardWrapper;
+import org.codemc.worldguardwrapper.region.IWrappedRegion;
 
 import java.util.HashMap;
 import java.util.UUID;
 
-public class RegionListener implements Listener{
+public class RegionListener implements Listener {
 
     private RegionManager utils;
-    @Getter private HashMap<UUID, String> playerInRegion = new HashMap<UUID, String>();
+    @Getter
+    private HashMap<UUID, String> playerInRegion = new HashMap<>();
 
-    public RegionListener(RegionManager utils){
+    public RegionListener(RegionManager utils) {
         this.utils = utils;
     }
 
-    @EventHandler
-    public void onJoin(PlayerJoinEvent event) {
-        //The from location has to be offset else the event will not be run
-        onMove(new PlayerMoveEvent(event.getPlayer(), event.getPlayer().getLocation().add(1, 0, 0), event.getPlayer().getLocation()));
-    }
-
-    @EventHandler
-    public void onTeleport(PlayerTeleportEvent event){
-        onMove(new PlayerMoveEvent(event.getPlayer(), event.getFrom(), event.getTo()));
-    }
-
-    @EventHandler
-    public void onMinecartMove(VehicleMoveEvent event) {
-        if (event.getVehicle().getPassenger() != null && event.getVehicle().getPassenger() instanceof Player) {
-            onMove(new PlayerMoveEvent((Player) event.getVehicle().getPassenger(), event.getFrom(), event.getTo()));
-        }
-    }
-
-    @EventHandler
-    public void onMove(PlayerMoveEvent e){
+    private void handleMovement(Player player, Location from, Location to) {
         //Only execute if the player moves an entire block
-        if(!(e.getFrom().getX() != e.getTo().getX() || e.getFrom().getZ() != e.getTo().getZ())) return;
+        if (from != null
+                && from.getBlockX() == to.getBlockX()
+                && from.getBlockY() == to.getBlockY()
+                && from.getBlockZ() != to.getBlockZ()) return;
```

Comment:
Shouldn't it be like this?
```suggestion
        if (from != null
                && from.getBlockX() == to.getBlockX()
                && from.getBlockY() != to.getBlockY()
                && from.getBlockZ() == to.getBlockZ()) return;
```

Reply:
Oh god, i tried to fix broken logic with another broken logic ahahahah

Reply:
It should be == in each axis :)

Reply:
and no, we NEED to check movements on the y axis as regions can cohexist at different Ys

Reply:
Yep that's true, then == everywhere 👍